pkg/virtctl/vnc/vnc.go

```diff
@@ -84,37 +84,51 @@ type VNC struct {
 	clientConfig clientcmd.ClientConfig
 }
 
-func (o *VNC) Run(cmd *cobra.Command, args []string) error {
-	namespace, _, err := o.clientConfig.Namespace()
+func (o *VNC) createLocalProxyForVmiWebsockConn(vmiName string) (*net.TCPListener, error) {
+	lnAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("127.0.0.1:%d", customPort))
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Can't resolve the address: %s", err.Error())
 	}
 
-	vmi := args[0]
+	ln, err := net.ListenTCP("tcp", lnAddr)
+	if err != nil {
+		return nil, fmt.Errorf("Can't listen on unix socket: %s", err.Error())
+	}
+
+	return ln, nil
+}
 
-	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+func (o *VNC) createVmiVncStream(vmiName string) (kubecli.StreamInterface, error) {
+	namespace, _, err := o.clientConfig.Namespace()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Cannot obtain KubeVirt client namespace: %v", err)
 	}
 
-	// setup connection with VM
-	vnc, err := virtCli.VirtualMachineInstance(namespace).VNC(vmi)
+	kubevirtClient, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
 	if err != nil {
-		return fmt.Errorf("Can't access VMI %s: %s", vmi, err.Error())
+		return nil, fmt.Errorf("Cannot obtain KubeVirt client: %v", err)
 	}
 
-	lnAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("127.0.0.1:%d", customPort))
+	vnc, err := kubevirtClient.VirtualMachineInstance(namespace).VNC(vmiName)
 	if err != nil {
-		return fmt.Errorf("Can't resolve the address: %s", err.Error())
+		return nil, fmt.Errorf("Can't access VMI %s: %s", vmiName, err.Error())
 	}
 
-	// The local tcp server is used to proxy the podExec websock connection to vnc client
-	ln, err := net.ListenTCP("tcp", lnAddr)
+	return vnc, nil
+}
+
+func (o *VNC) Run(cmd *cobra.Command, args []string) error {
+	vmiName := args[0]
+
+	vnc, err := o.createVmiVncStream(vmiName)
 	if err != nil {
-		return fmt.Errorf("Can't listen on unix socket: %s", err.Error())
+		return err
+	}
+
+	ln, err := o.createLocalProxyForVmiWebsockConn(vmiName)
```

Comment:
Could you change this to `createLocalProxyForVmiWebsocketConn`.
Websocket is the word used mostly throughout the codebase and if somebody would search for it Websock would not show up.